cql3/expr/expression.cc

```diff
@@ -225,14 +221,28 @@ static managed_bytes_opt get_value(const column_maybe_subscripted& col, const co
                     format("Tried to evaluate expression with wrong type for subscript of {}",
                         cdef->name_as_text()));
             }
+          if (col_type->is_map()) {
+            const auto& data_map = value_cast<map_type_impl::native_type>(deserialized);
             const auto found = key.view().with_linearized([&] (bytes_view key_bv) {
                 using entry = std::pair<data_value, data_value>;
                 return std::find_if(data_map.cbegin(), data_map.cend(), [&] (const entry& element) {
                     return key_type->compare(element.first.serialize_nonnull(), key_bv) == 0;
                 });
             });
             return found == data_map.cend() ? std::nullopt : managed_bytes_opt(found->second.serialize_nonnull());
-        }
+          } else if (col_type->is_list()) {
+            const auto& data_list = value_cast<list_type_impl::native_type>(deserialized);
+            auto key_deserialized = key.view().with_linearized([&] (bytes_view key_bv) {
+                return key_type->deserialize(key_bv);
+            });
+            auto key_int = value_cast<int32_t>(key_deserialized);
+            if (key_int < 0 || size_t(key_int) >= data_list.size()) {
+                return std::nullopt;
+            }
+            return managed_bytes_opt(data_list[key_int].serialize_nonnull());
+          } else
```

Comment:
Something's extremely off with braces here and indentation later... I think it only works because the `else` body is a one-liner and you skipped a brace+indentation below.

Reply:
Ouch! will fix.

Reply:
Fixed. I remember feeling something was odd.